nexus/deployment/src/planner.rs

```diff
@@ -133,7 +133,30 @@ impl<'a> Planner<'a> {
                 }
             }
 
-            if ncrucibles_added > 0 {
+            // TODO XXX To test Nexus zone addition, attempt to ensure every
+            // sled has a Nexus zone. This is NOT a sensible policy, and should
+            // be removed before this lands on `main`.
```

Comment:
(discussed offline) we should probably do _something_ about this, even if it's just to change the comment.

Reply:
Yep. Trying to briefly summarize the offline discussion, as I'm working on changes here that are nontrivial:

* Instead of adding a Nexus to every sled, we'll try to ensure there are 3 Nexuses. (For testing on madrid, we can easily build a local version that wants 4 instead of 3.)
* If it needs to add new Nexuses, the planner should prioritize one of the sleds with the minimum number of Nexus zones. If there are more sleds than Nexuses (the typical case), it'll pick sleds with no Nexus. If there are fewer than 3 sleds (common in dev/test), it'll be forced to assign multiple Nexus instances to one sled.